pennylane/ops/functions/equal.py

```diff
@@ -86,8 +118,13 @@ def equal(
         True
     """
 
-    if not isinstance(op2, type(op1)):
-        return False
+    # types don't have to be the same type, they just both have to be Observables
+    if isinstance(op1, Observable) and isinstance(op2, Observable):
+        pass
+    # for everything else, types must match
+    else:
+        if not isinstance(op2, type(op1)):
+            return False
```

Comment:
```suggestion
    if not isinstance(op2, type(op1)) and not isinstance(op1, Observable):
            return False
```

Reply:
So I committed this change, and then ran my example notebook and remembered why it wasn't like this. For the behaviour to be consistent regardless of input order, we need to check that they are _both_ Observables. Otherwise you get different behaviour depending on the order of the arguments if you compare a Pauli operator (which `singledispatch` considers first and foremost an Observable and secondarily an Operation) with an `Operation`. 

If the `Operation` is `op1`, it concludes they don't match and returns False without ever calling _equal. But if the Pauli operator is op1, it calls _equals_observables, which expects to Observables, and raises an attribute error:

```
op1 = qml.PauliX(0)
op2 = qml.RX(np.pi/2, 0)

qml.equal(op2, op1)
>>> False

qml.equal(op1, op2) 
>>> Raises AttributeError: 'RX' object has no attribute '_obs_data'
```

Is there a more concise way to phrase this that still confirms that both `op1` and `op2` are `Observable` objects?

Also I suppose I should add a check that input order doesn't matter to the tests.